src/Umbraco.Cms.Api.Management/Services/FileSystem/FileSystemTreeServiceBase.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Umbraco.Cms.Api.Management.Extensions;
+using Umbraco.Cms.Api.Management.ViewModels.FileSystem;
+using Umbraco.Cms.Api.Management.ViewModels.Tree;
+using Umbraco.Cms.Core.IO;
+using Umbraco.Extensions;
+
+namespace Umbraco.Cms.Api.Management.Services.FileSystem;
+
+public abstract class FileSystemTreeServiceBase : IFileSystemTreeService
+{
+    protected abstract IFileSystem FileSystem { get; }
+
+    public FileSystemTreeItemPresentationModel[] GetAncestorModels(string path, bool includeSelf)
+    {
+        var directories = path.Split(Path.DirectorySeparatorChar).Take(Range.EndAt(Index.FromEnd(1))).ToArray();
+        var result = directories
+            .Select((directory, index) => MapViewModel(string.Join(Path.DirectorySeparatorChar, directories.Take(index + 1)), directory, true))
+            .ToList();
+
+        if (includeSelf)
+        {
+            var selfIsFolder = FileSystem.FileExists(path) is false;
+            result.Add(MapViewModel(path, GetFileSystemItemName(selfIsFolder, path), selfIsFolder));
+        }
+
+        return result.ToArray();
+    }
+
+    public FileSystemTreeItemPresentationModel[] GetPathViewModels(string path, int skip, int take, out long totalItems)
+    {
+        path = path.VirtualPathToSystemPath();
+        var allItems = GetDirectories(path)
+            .Select(directory => new { Path = directory, IsFolder = true })
+            .Union(GetFiles(path).Select(file => new { Path = file, IsFolder = false }))
+            .ToArray();
+
+        totalItems = allItems.Length;
+
+        FileSystemTreeItemPresentationModel ViewModel(string itemPath, bool isFolder)
+            => MapViewModel(
+                itemPath,
+                GetFileSystemItemName(isFolder, itemPath),
+                isFolder);
+
+        return allItems
+            .Skip(skip)
+            .Take(take)
+            .Select(item => ViewModel(item.Path, item.IsFolder))
+            .ToArray();
+    }
+
+    public FileSystemTreeItemPresentationModel[] GetSiblingsViewModel(string path, int before, int after, out long totalBefore, out long totalAfter)
```

Comment:
Suggest this should be `GetSiblingsViewModels` to align with other methods that are pluralised.